src/transformers/trainer_seq2seq.py

```diff
@@ -0,0 +1,247 @@
+# Copyright 2020 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import warnings
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import torch
+from packaging import version
+from torch import nn
+from torch.utils.data import DistributedSampler, RandomSampler
+from torch.utils.data.dataset import Dataset
+
+from .file_utils import is_torch_tpu_available
+from .trainer import Trainer
+from .trainer_pt_utils import LabelSmoother, get_tpu_sampler
+from .trainer_utils import PredictionOutput
+from .training_args import ParallelMode
+from .utils import logging
+
+
+if version.parse(torch.__version__) >= version.parse("1.6"):
+    from torch.cuda.amp import autocast
+
+
+logger = logging.get_logger(__name__)
+
+
+class Seq2SeqTrainer(Trainer):
+    def __init__(self, *args, ignore_pad_token_for_loss=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        # TODO: When BART uses -100 everywhere, this can be removed entirely.
```

Comment:
BART now works with -100, the issue was with label smoothing (which you fixed) and the `shift_tokens_right` function, which is fixed in #9135

Reply:
From the `src/transformers` side we can completely remove the deprecation warning and `ignore_pad_token_for_loss` arg IMO. On the other hand, it seemed like `ignore_pad_token_for_loss` was the default case for all training code in `examples/seq2seq` -> so might be good to keep it as it is now for some time

Reply:
Will adapt the TODO and the warning then, to check all mentions in the examples. Once we're sure it's not used anywhere, will remove the argument. but let's try to have it removed before the next release, so it's never a real "breaking change" to remove that arg.

Reply:
On second thoughts, will have to change all scripts for the `max_target_length` -> `max_length` so I'll remove those `ignore_pad_token_for_loss` in passing. (just realized it is never actually used inside the code and the tests are still passing.)